drivebrain_app/DriveBrainApp.cpp

```diff
@@ -0,0 +1,239 @@
+// DriveBrainApp.cpp
+#include "DriveBrainApp.hpp"
+
+std::atomic<bool> DriveBrainApp::stop_signal_{false};
+
+std::string DriveBrainApp::getParamPathFromArgs(int argc, char* argv[]) {
+    namespace po = boost::program_options;
+    po::options_description desc("Allowed options");
+    std::string param_path;
+    
+    desc.add_options()
+        ("param-path,p", po::value<std::string>(&param_path)->default_value("config/drivebrain_config.json"), 
+         "Path to the parameter JSON file");
+
+    po::variables_map vm;
+    po::store(po::command_line_parser(argc, argv).options(desc).allow_unregistered().run(), vm);
+    po::notify(vm);
+    
+    return param_path;
+}
+
+void DriveBrainApp::parseCommandLine(int argc, char* argv[]) {
+    namespace po = boost::program_options;
+    po::options_description desc("Allowed options");
+    
+    desc.add_options()
+        ("help,h", "produce help message")
+        ("dbc-path,d", po::value<std::string>(), "Path to the DBC file (optional)");
+
+    po::variables_map vm;
+    po::store(po::command_line_parser(argc, argv).options(desc).allow_unregistered().run(), vm);
+    po::notify(vm);
+    
+    if (vm.count("help")) {
+        std::stringstream ss;
+        ss << desc;
+        spdlog::info("{}", ss.str());
+        throw std::runtime_error("Help requested");
+    }
+    
+    if (vm.count("dbc-path")) {
+        dbc_path_ = vm["dbc-path"].as<std::string>();
+    }
+}
+
+DriveBrainApp::DriveBrainApp(int argc, char* argv[])
+    : param_path_(getParamPathFromArgs(argc, argv))
+    , logger_(core::LogLevel::INFO)
+    , config_(param_path_)
+{
+    parseCommandLine(argc, argv);
+    setupSignalHandler();
+    spdlog::set_level(spdlog::level::warn);
+}
+
+DriveBrainApp::~DriveBrainApp() {
+    stop();
+}
+
+void DriveBrainApp::setupSignalHandler() {
+    std::signal(SIGINT, [](int signal) {
+        spdlog::info("Interrupt signal ({}) received. Cleaning up...", signal);
+        DriveBrainApp::stop_signal_.store(true);
+    });
+}
```

Comment:
instead of making this a member function, for signal handling I would suggest leaving this as a part of the regular `main.cpp` as this isnt crucial for the drivebrain runtime construction itself and is can be left as a separate implementation in the `main.cpp` outside of the `DriveBrainApp` class. it would instead communicate for destruction of the `DriveBrainApp` class instance.